poo/abstraction/en/Abstraction.md

```diff
@@ -0,0 +1,117 @@
+# Abstraction
+
+The **abstraction** idea, is chopper a class in parts, to be able to share (via a parameter passed in method for example) **an only** of this part rather the object “in full”.
+To understand the utility of this principle and how apply, take for example a class Human, since this is a thing that we know better.
+
+```java
+
+class Human {
+
+    int air;
+    int satiety;
+    float energy;
+
+
+    void breathe() {
+        air += 10;
+    }
+
+    void eat() {
+        satiety += 5;
+    }
+
+    void move() {
+        System.out.println("Un humain souhaite se déplacer !");
+    }
+
+    void consumeEnergy(float value) {
+        this.energie -= value;
+    }
+}
+```
+
+Now, imagine that we have a class Hiking which allows organise a walk enter Humans. A thing of style:
+
+```java
+class Hiking {
+
+    List<Human> participants;
+
+    public Hiking() { this.participants = new ArrayList<>(); }
+
+    void addParticipant(Human participant) { this.participants.add(participant); }
+
+    void moveAll() {
+        for(Human human : participants) {
+            human.move();
+            human.consumeEnergy(3);
+        }
+    }
+}
+```
+
+So far so good, we have that we want. This code works perfectly, except, in reality, he contain two major problems.

```

Comment:
Bad pronoun and conjugation:
- `This code [...] he contain [...]` -> `This code [...] it contains [...]`
```suggestion
So far so good, we have that we want. This code works perfectly, except, in reality, it contains two major problems.

```

Reply:
Corrected, tanks!